src/main/java/gregtech/common/covers/detector/CoverDetectorItemAdvanced.java

```diff
@@ -0,0 +1,169 @@
+import gregtech.api.capability.impl.FilteredItemHandler;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.CoverWithUI;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.util.GTLog;
+import gregtech.api.util.GTUtility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.common.covers.filter.ItemFilterContainer;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.*;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+
+import java.util.regex.Pattern;
+
+public class CoverDetectorItemAdvanced extends CoverDetectorItem implements CoverWithUI {
+    private final int DEFAULT_MIN = 64;
+    private final int DEFAULT_MAX = 512;
+    private int min, max;
+    protected ItemFilterContainer itemFilter;
+
+    public CoverDetectorItemAdvanced(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.itemFilter = new ItemFilterContainer(this);
+        this.min = DEFAULT_MIN;
+        this.max = DEFAULT_MAX;
+    }
+
+    @Override
+    public void renderCover(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline, Cuboid6 plateBox, BlockRenderLayer layer) {
+        // todo replace with unique texture
+        Textures.DETECTOR_ITEM.renderSided(attachedSide, plateBox, renderState, pipeline, translation);
+    }
+
+    @Override
+    public ModularUI createUI(EntityPlayer player) {
+        int PADDING = 3;
+        int SIZE = 18;
+
+        WidgetGroup group = new WidgetGroup();
+        group.addWidget(new LabelWidget(10, 8, "cover.advanced_item_detector.label"));
+
+        // set min fluid amount
+        group.addWidget(new LabelWidget(10, 5 + (SIZE + PADDING), "cover.advanced_item_detector.min"));
+        group.addWidget(new ImageWidget(98 - 4, (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
+        group.addWidget(new TextFieldWidget2(98, 5 + (SIZE + PADDING), 4 * SIZE, SIZE,
+                this::getMinValue, this::setMinValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
+        );
+
+        // set max fluid amount
+        group.addWidget(new LabelWidget(10, 5 + 2 * (SIZE + PADDING), "cover.advanced_item_detector.max"));
+        group.addWidget(new ImageWidget(98 - 4, 2 * (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
+        group.addWidget(new TextFieldWidget2(98, 5 + 2 * (SIZE + PADDING), 4 * SIZE, SIZE,
+                this::getMaxValue, this::setMaxValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
```

Comment:
See my other comment about the regex pattern.

Reply:
resolved as of https://github.com/GregTechCEu/GregTech/pull/1448/commits/bdd0b85af92f843d04c8b77e6cc50fa68c159bf0